packages/core/test/actions.test.ts

```diff
@@ -3664,3 +3664,224 @@ it('should call a referenced action responding to an initial raise with updated
+    const machine = createMachine(
+      {
+        context,
+        on: {
+          event: { actions: 'group' }
+        }
+      },
+      {
+        actions: {
+          group: [
+            ({ context }) => action1(context),
+            ({ event }) => action2(event)
+          ]
+        }
+      }
+    );
+
+    const service = interpret(machine).start();
+
+    service.send(event);
+
+    expect(action1).toHaveBeenCalledWith(context);
+    expect(action2).toHaveBeenCalledWith(event);
+  });
+
+  it('should execute a mix of action types in a group', () => {
+    const action1 = jest.fn();
+    const action2 = jest.fn();
+
+    const machine = createMachine(
+      {
+        entry: 'group',
+        context: {
+          value: 'some value'
+        }
+      },
+      {
+        actions: {
+          group: [
+            'action1',
+            assign({
+              value: ({ context }) => `${context.value}!`
+            }),
+            { type: 'action2' }
+          ],
+          action1: ({ context }) => action1(context.value),
+          action2: ({ context }) => action2(context.value)
+        }
+      }
+    );
+
+    interpret(machine).start();
+
+    expect(action1).toHaveBeenCalledWith('some value');
+    expect(action2).toHaveBeenCalledWith('some value!');
+  });
+
+  it('should be able to reference groups from other groups while respecting action order', () => {
+    const action = jest.fn();
+
+    const machine = createMachine(
+      {
+        entry: 'group1'
+      },
+      {
+        actions: {
+          group1: ['group2', 'group3'],
+          group2: [assign({ value: 'a' }), 'action', 'group4'],
+          group3: [assign({ value: 'b' }), 'action'],
+          group4: [assign({ value: 'c' }), 'action'],
+          action: ({ context }) => action(context.value)
+        }
+      }
+    );
+
+    interpret(machine).start();
+
+    expect(action).toHaveBeenCalledTimes(3);
+    expect(action).toHaveBeenNthCalledWith(1, 'a');
+    expect(action).toHaveBeenNthCalledWith(2, 'c');
+    expect(action).toHaveBeenNthCalledWith(3, 'b');
+  });
+
+  it('should pass params to executed group actions', () => {
+    const handler1 = jest.fn();
+    const handler2 = jest.fn();
+
+    const machine = createMachine(
+      {
+        entry: {
+          type: 'messageHandlers',
+          params: {
+            message: 'Hello world!'
+          }
+        }
+      },
+      {
+        actions: {
+          messageHandlers: [
+            'handler1',
+            { type: 'otherHandlers', params: { level: 'info' } }
+          ],
+          otherHandlers: ['handler2'],
+          handler1: ({ action }) => handler1(action.params),
+          handler2: ({ action }) => handler2(action.params)
+        }
+      }
+    );
+
+    interpret(machine).start();
+
+    expect(handler1).toHaveBeenCalledWith({ message: 'Hello world!' });
+    expect(handler2).toHaveBeenCalledWith({
+      message: 'Hello world!',
+      level: 'info'
+    });
```

Comment:
This has an implicit test for `params` overriding. Should we be more explicit with a separate test?

Reply:
Might not be a need because if the behavior changes, this test will fail.

cc. @Andarist for his thoughts